pex/resolve/lockfile/__init__.py

```diff
@@ -155,6 +158,13 @@ def parse_lockable_requirements(
             )
         )
 
+    if not parsed_requirements:
+        return Error(
+            "No requirements requested. This should not be possible: either users will have "
+            "specified requirements on the command line, or we will fall back to resolving the "
+            "entire lockfile. Please file a bug at https://github.com/pantsbuild/pex/issues/new."
```

Comment:
Some defensive programming - note that Pex already complains if you use a lock with no requirements. But I figure this is better than the `ZeroDivisionError` from https://github.com/pantsbuild/pex/issues/1657.